tests/lib/rules/template-tag-spacing.js

```diff
@@ -0,0 +1,259 @@
+ * @author Jonathan Wilsson
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/template-tag-spacing"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 6 } });
+
+ruleTester.run("template-tag-spacing", rule, {
+    valid: [
+        "tag`name`",
+        { code: "tag`name`", options: ["never"] },
+        { code: "tag `name`", options: ["always"] },
+        "tag`hello ${name}`",
+        { code: "tag`hello ${name}`", options: ["never"] },
+        { code: "tag `hello ${name}`", options: ["always"] },
+        "tag/*here's a comment*/`Hello world`",
+        { code: "tag/*here's a comment*/`Hello world`", options: ["never"] },
+        { code: "tag /*here's a comment*/`Hello world`", options: ["always"] },
+        { code: "tag/*here's a comment*/ `Hello world`", options: ["always"] },
+        "new tag`name`",
+        { code: "new tag`name`", options: ["never"] },
+        { code: "new tag `name`", options: ["always"] },
+        "new tag`hello ${name}`",
+        { code: "new tag`hello ${name}`", options: ["never"] },
+        { code: "new tag `hello ${name}`", options: ["always"] },
+        "(tag)`name`",
+        { code: "(tag)`name`", options: ["never"] },
+        { code: "(tag) `name`", options: ["always"] },
+        "(tag)`hello ${name}`",
+        { code: "(tag)`hello ${name}`", options: ["never"] },
+        { code: "(tag) `hello ${name}`", options: ["always"] },
+        "new (tag)`name`",
+        { code: "new (tag)`name`", options: ["never"] },
+        { code: "new (tag) `name`", options: ["always"] },
+        "new (tag)`hello ${name}`",
+        { code: "new (tag)`hello ${name}`", options: ["never"] },
+        { code: "new (tag) `hello ${name}`", options: ["always"] }
+    ],
+    invalid: [
+        {
+            code: "tag `name`",
+            output: "tag`name`",
+            errors: [
+                { message: "Unexpected space between template tag and template literal." }
+            ]
+        },
+        {
+            code: "tag `name`",
+            output: "tag`name`",
+            errors: [
+                { message: "Unexpected space between template tag and template literal." }
+            ],
+            options: ["never"]
+        },
+        {
+            code: "tag`name`",
+            output: "tag `name`",
+            errors: [
+                { message: "Missing space between template tag and template literal." }
+            ],
+            options: ["always"]
+        },
+        {
+            code: "tag /*here's a comment*/`Hello world`",
```

Comment:
Thanks for updating this! One final thing - this still gets autofixed so we do need an `output` property (even if it's just to assert that it doesn't get fixed).  With the current logic, I believe it will remove the comment, correct? My preference would be to either figure out a way to keep the comment in the fixed output or to not fix if a comment exists (with a strong preference for the former), rather than simply remove the comment. Thoughts? 

@not-an-aardvark Can you be a resource here? I think you've written more autofixes than the rest of us combined by now!